### PR TITLE
Tpt3_102 rule00738

### PR DESCRIPTION
**Pull request by dmytro-palczewski-osi**
rule R-00738 implementation

### CODE REVIEW

File: api-core-test-util/src/main/java/gov/ca/cwds/util/TestUtils.java [hunk 1]
```diff
@@ -7,8 +7,9 @@
 
   public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd");
 
+  private TestUtils(){}
+
   public static LocalDate localDate(String dateStr) {
     return LocalDate.parse(dateStr, DATE_FORMATTER);
```
Comment (sorych) [inline api-core-test-util/src/main/java/gov/ca/cwds/util/TestUtils.java]: if we use it this way for test purposes why can't we just use LocalDate method of(int year, Month month, int dayOfMonth)?

Comment (dmytro-palczewski-osi) [reply api-core-test-util/src/main/java/gov/ca/cwds/util/TestUtils.java]: TestUtils class is removed

File: legacy-data-access-services/src/main/resources/client-legacy-rules/R-00738.drl [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import gov.ca.cwds.cms.data.access.dto.ClientEntityAwareDTO;
+
+global java.util.Set validationDetailsList
+
+rule "R-00738"
+  dialect "mvel"
+  agenda-group "client-agenda"
+  when
+      /*
+        Code: R - 00741
```
Comment (sorych) [inline legacy-data-access-services/src/main/resources/client-legacy-rules/R-00738.drl]: 738?

File: legacy-data-access-services/src/test/java/gov/ca/cwds/cms/data/access/service/impl/R00738Test.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  @Test
+  public void testBirthDateNull() throws Exception {
+    Client client = new Client();
+    client.setCreationDate(LocalDate.of(2018, 1, 3));
+    client.setBirthDate(null);
+
+    checkRuleSatisfied(client, RULE_NAME);
+  }
```
Comment (denys-davydov) [inline legacy-data-access-services/src/test/java/gov/ca/cwds/cms/data/access/service/impl/R00738Test.java]: I would add a test on the equal creation and birth dates to cover the corner case.

Comment (dmytro-palczewski-osi) [reply legacy-data-access-services/src/test/java/gov/ca/cwds/cms/data/access/service/impl/R00738Test.java]: it exists, see testCreationDateEqBirthDate()( method
